src/engine/Timer.ts

```diff
@@ -17,16 +23,20 @@ export class Timer {
   private _totalTimeAlive: number = 0;
   private _paused: boolean = false;
   private _numberOfTicks: number = 0;
+  private _callbacks: { [key: string]: () => void | undefined };
   public complete: boolean = false;
   public scene: Scene = null;
 
   /**
-   * @param fcn        The callback to be fired after the interval is complete.
    * @param interval   Interval length
+   * @param options    Options - repeats, numberOfRepeats, fcn
    * @param repeats    Indicates whether this call back should be fired only once, or repeat after every interval as completed.
    * @param numberOfRepeats Specifies a maximum number of times that this timer will execute.
+   * @param fcn        The callback to be fired after the interval is complete.
    */
-  constructor(fcn: () => void, interval: number, repeats?: boolean, numberOfRepeats?: number) {
+  constructor(interval: number, options: TimerOptions = {}) {
```

Comment:
Let's go ahead and make `interval` part of `TimerOptions` as well. We can add a constructor overload to preserve backwards compatibility

https://github.com/excaliburjs/Excalibur/issues/396#issuecomment-538583725

Reply:
Totally agree with that. Proceeding with the change :)